src/observability.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the Observability class to represent the observability stack for Synapse."""
+
+
+import ops
+from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
+from charms.loki_k8s.v1.loki_push_api import LogProxyConsumer
+from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
+
+import wazuh
+
+LOG_PATHS = [(wazuh.LOGS_PATH / "*.log").absolute().as_posix()]
+PROMETHEUS_PORT = 5000
+
+
+class Observability:  # pylint: disable=too-few-public-methods
+    """A class representing the observability stack for Synapse application."""
+
+    def __init__(self, charm: ops.CharmBase):
+        """Initialize a new instance of the Observability class.
+
+        Args:
+            charm: The charm object that the Observability instance belongs to.
+        """
+        self._grafana_dashboards = GrafanaDashboardProvider(
+            charm, relation_name="grafana-dashboard"
+        )
+        prometheus_target = [f"*:{PROMETHEUS_PORT}"]
+        self._metrics_endpoint = MetricsEndpointProvider(
+            charm,
+            relation_name="metrics-endpoint",
+            jobs=[
+                {
+                    "job_name": "stats_exporter",
+                    "static_configs": [{"targets": prometheus_target}],
+                },
+            ],
+        )
+        self._logging = LogProxyConsumer(
+            charm,
+            relation_name="logging",
+            logs_scheme={
+                f"{wazuh.CONTAINER_NAME}": {
+                    "log-files": LOG_PATHS,
+                },
+            },
+        )
```

Comment:
Can't we use Pebble log-forwarding feature instead of the old "fetch a file" approach ?
Ideally, if possible, Wazuh should write its logs to stdout/stderr and then we configure pebble to forward them

Reply:
Had a quick look and it seems that wazuh server is not able to write logs to stdout/stderr

Reply:
Indeed. I haven't found out how either